src/freertos_drivers/esp32/Esp32WiFiConfiguration.hxx

```diff
@@ -0,0 +1,139 @@
+
+    /// Visible description for the hub enable field.
+    static constexpr const char *HUB_ENABLE_DESC = 
+        "Defines this node as a hub which can accept connections";
+
+    /// Visible name for the hub_listener_port field.
+    static constexpr const char *HUB_LISTENER_PORT_NAME =
+        "Hub Listener Port";
+
+    /// Visible name for the hub_listener_port field.
+    static constexpr const char *HUB_LISTENER_PORT_DESC = 
+        "Defines the TCP/IP listener port this node will use when operating "
+        "as a hub. Most of the time this does not need to be changed.";
+
+    /// Visible name for the link_config group.
+    static constexpr const char *UPLINK_NAME = "Node Uplink Configuration";
+
+    /// Visible name for the link_config group.
+    static constexpr const char *UPLINK_DESC =
+        "Configures how this node will connect to other nodes.";
+};
+
+/// CDI Configuration for an @ref Esp32WiFiManager managed hub.
+CDI_GROUP(HubConfiguration);
+/// Allows the node to become a Grid Connect Hub.
+CDI_GROUP_ENTRY(enable, openlcb::Uint8ConfigEntry,
+    Name(Esp32WiFiConfigurationParams::HUB_ENABLE_NAME),
+    Description(Esp32WiFiConfigurationParams::HUB_ENABLE_DESC),
+    Min(0), Max(1), Default(0),
+    MapValues(Esp32WiFiConfigurationParams::BOOLEAN_MAP));
+/// Specifies the port which should be used by the hub.
+CDI_GROUP_ENTRY(port, openlcb::Uint16ConfigEntry,
+    Name(Esp32WiFiConfigurationParams::HUB_LISTENER_PORT_NAME),
+    Description(Esp32WiFiConfigurationParams::HUB_LISTENER_PORT_DESC),
+    Min(1), Max(65535),
+    Default(openlcb::TcpClientDefaultParams::DEFAULT_PORT))
+/// Specifies the mDNS service name to advertise for the hub.
+CDI_GROUP_ENTRY(service_name, openlcb::StringConfigEntry<48>,
+    Name(openlcb::TcpClientDefaultParams::SERVICE_NAME),
+    Description(openlcb::TcpClientDefaultParams::SERVICE_DESCR));
+CDI_GROUP_END();
+
+/// CDI Configuration for an @ref Esp32WiFiManager managed node.
+CDI_GROUP(WiFiConfiguration);
+/// Allows the WiFi system to use power-saving techniques to conserve power
+/// when the node is powered via battery.
+CDI_GROUP_ENTRY(sleep, openlcb::Uint8ConfigEntry,
+    Name(Esp32WiFiConfigurationParams::WIFI_POWER_SAVE_NAME),
+    Description(Esp32WiFiConfigurationParams::WIFI_POWER_SAVE_DESC),
+    Min(0), Max(1), Default(0),
+    MapValues(Esp32WiFiConfigurationParams::BOOLEAN_MAP));
+/// CDI Configuration to enable this node to be a hub.
+CDI_GROUP_ENTRY(hub, HubConfiguration,
+    Name(Esp32WiFiConfigurationParams::HUB_NAME),
+    Description(Esp32WiFiConfigurationParams::HUB_DESC));
+/// CDI Configuration for this node's connection to an uplink hub.
+CDI_GROUP_ENTRY(uplink,
+    openlcb::TcpClientConfig<openlcb::TcpClientDefaultParams>,
+    Name(Esp32WiFiConfigurationParams::UPLINK_NAME),
+    Description(Esp32WiFiConfigurationParams::UPLINK_DESC));
+CDI_GROUP_END();
+
+} // namespace openmrn_arduino
+
+using openmrn_arduino::WiFiConfiguration;
```

Comment:
I don't think this is necessary.
We should only use using in .hxx files for stuff that the end-user will have to write in their sketch in a place where a namespace designation would be surprising or ugly.

Reply:
will remove, I wasn't sure if it was needed and erred on the side of caution.

Reply:
Turns out this is required after all. With the updates being done for the remaining comments below this wouldn't compile without the using declaration as it is used in the config.h.